google-pub/c/README.md

```diff
@@ -1,55 +1,48 @@
+# Google Cloud MQTT Sample
+
+## Introduction
+This sample app demonstrates connecting to Google Cloud IoT Core and publishing a single message. This Sample is modified for ease of creating a new Intel(R) System Studio project.
+
+## Hardware requirements
+No special H/W requirements, as long as your device could be connected to the internet.
+
+## Software requirements
+This version of the sample has been tested on Ubuntu Linux.
+
+## Setup
+Refer to [this document](https://cloud.google.com/iot/docs/quickstart) on how to create deviceid, registryid, and projectid parameters as they are used in calculating the client ID used for connecting to the MQTT bridge.
 
 # Building
-To build the sample, you must first download, build, and install the following
-projects:
+To build the sample, you must first download, build, and install the following projects, you can either install from individual links or follow the below commands.
 
 * [Jansson](https://github.com/akheron/jansson)
+	git clone https://github.com/akheron/jansson
+	cd jansson && cmake . && make && make install
 * [libjwt](https://github.com/benmcollins/libjwt)
+	add-apt-repository -y ppa:ben-collins/libjwt && apt-get update
+	apt-get install -y libjwt0
 * [OpenSSL](https://github.com/openssl/openssl)
 * [paho.mqtt.c](https://github.com/eclipse/paho.mqtt.c)
-* [Check](https://github.com/libcheck/check) (Optional, used for build tests)
+	apt-get install -y gcc
+	apt-get install -y libssl-dev
+	curl -OJL https://github.com/eclipse/paho.mqtt.c/archive/v1.2.0.tar.gz
+	tar xf paho.mqtt.c-1.2.0.tar.gz
+	cd paho.mqtt.c-1.2.0
+	prefix=/usr make -j4 install
 
 After you have installed all the dependencies, you can build the sample as:
 
     gcc mqtt_ciotc.c -lssl -lcrypto -lpaho-mqtt3cs -ljwt -o mqtt_ciotc
 
-or by invoking `make`.
-
-Due to conflicting libraries existing on your build machine, you may need to
-specify the library paths, for example, if you installed the dependencies in
-$HOME/code, you could do the following to help the linker find the correct
-libraries:
-
-    gcc mqtt_ciotc.c -L$HOME/code/openssl -lssl -lcrypto -lpaho-mqtt3cs -L$HOME/code/libjwt -ljwt -o mqtt_ciotc
-
-For simplifying the build process, a Docker image is provided that generates
-the build configuration. After installing Docker, browse to the `docker` folder
-and run:
-
-    docker build -t your-docker-tag .
-
-It will take a little while for the Docker image to build. After the Docker
-image is built, you can access the builder as:
-
-    docker run -t -i your-docker-tag /bin/bash
-
 # Running
-After you have successfully built the sample, you are almost ready to run the
-sample.
+After you have successfully built the sample, you are almost ready to run the sample.
 
-1. Setup your libraries:
+## Setup your libraries:
 
     export LD_LIBRARY_PATH=/usr/local/lib
     export LD_RUN_PATH=/usr/local/lib
 
-2. Browse to the sample folder
-
-    cd mqtt-ciotc
-
-3. Run the sample
+## Run the sample
```

Comment:
The ISS readmes have header called 'SETUP'. Please show that add the argument value at project creation or added via the header files. 

Can the user provide command line arguments in ISS? If no, this should all go in a 'Building and Running sample outside ISS'

Reply:
I don't think user can provide command line arguments in ISS. I can add those steps in a. separate section.